syz-manager/rpc.go

```diff
@@ -327,68 +332,27 @@ func (serv *RPCServer) ExchangeInfo(a *rpctype.ExchangeInfoRequest, r *rpctype.E
 		return nil
 	}
 
-	if !serv.checkDone.Load() {
-		serv.mu.Lock()
-		if !serv.checkDone.Load() {
-			serv.checkResults = append(serv.checkResults, a.Results...)
-			if len(serv.checkResults) < serv.needCheckResults {
-				numRequests := min(len(serv.checkProgs), a.NeedProgs)
-				r.Requests = serv.checkProgs[:numRequests]
-				serv.checkProgs = serv.checkProgs[numRequests:]
-			} else {
-				if err := serv.finishCheck(); err != nil {
-					log.Fatalf("check failed: %v", err)
-				}
-				serv.checkProgs = nil
-				serv.checkResults = nil
-				serv.checkFiles = nil
-				serv.checkFilesInfo = nil
-				serv.checkDone.Store(true)
-			}
+	// First query new inputs and only then post results.
```

Comment:
Depending on the number of dependent request iterations, we'd indeed see some empty exchange requests. But
1) There don't seem to be reasons why we'd have too many of them.
2) We always have at least several VMs, it will smooth the effect out.

Overall, I'd say that the order mostly affects `triageJob.deflake()` requests -- we'd like to maximize the probability that they end up on different VMs. In all other parts of `pkg/fuzzer`, it should not really matter.

Theoretically we can explicitly ensure this behavior (provided there are enough VMs) and then it won't matter whether we process requests or results first.

Reply:
I started looking into switching manager<->fuzzer communication to flatrpc and a single TCP connection. When we do that, this aspect won't matter so I think we can ignore it for now.
In the new scheme, fuzzer sends "I started executing a request", after this manager sends a new request to the fuzzer, then fuzzer sends "I finished executing the previous request".
So the order you want will happen naturally since we send a new request before the previous has completed.